dfastmi/gui.py

```diff
@@ -375,10 +377,11 @@ def update_qvalues(reach:Reach) -> None:
             dialog[prefix+"qval"].setText(qval)
             tabs.setTabText(1+j,qval+" m3/s")
             
-    try:
-        nwidth = reach.normal_width
+    try:        
         q_threshold = float(dialog["qthr"].text())
-        [_, _, time_mi, _, _, _, celerity] = dfastmi.batch.core.get_levels_v2(reach, q_threshold, nwidth)
+        time_fractions_of_the_year = ConfigurationInitializer.get_time_fractions_of_the_year(reach.hydro_t)
+        time_mi = ConfigurationInitializer.calculate_time_mi(q_threshold, reach.hydro_q, time_fractions_of_the_year)
+        celerity = ConfigurationInitializer.get_bed_celerity(reach, reach.hydro_q)
```

Comment:
Should calculations be the responsibility of a `ConfigurationInitializer`?

Reply:
Discussed, needed to configure the real settings for the analysis